scipp-imaging-2D.ipynb

```diff
@@ -446,9 +915,61 @@
    },
    "outputs": [],
    "source": [
-    "fit_ds = sc.compat.mantid.fit(ws, \n",
-    "                     function='name=LinearBackground,A0=5000,A1=0;name=UserFunction,Formula=h*erfc(a*(x-x0)),h=5000,a=-0.5,x0=4',\n",
-    "                    workspace_index=2, start_x=3.6, end_x=4.4)"
+    "def write_ws_to_ascii(input_list, output_filename_stem, output_directory):\n",
+    "    '''\n",
+    "    Takes histogram from mantid workspace and writes an ASCII file.\n",
+    "    New filename will be the filename of the input workspace and\n",
+    "    a given extension appended.\n",
+    "\n",
+    "    :param input_ws: input workspace from mantid\n",
+    "    :param output_directory: Where to write the new file\n",
+    "    :param output_extension: which extension should the ASCII file have\n",
+    "\n",
+    "    :raises: Error if file exists\n",
+    "    '''\n",
+    "\n",
+    "    output_filenames = []\n",
+    "    for i, li in enumerate(input_list):\n",
+    "        bragg_edge = li[0][2].value\n",
+    "        output_filename = \"_{}_{:.3f}A_{}.txt\".format(output_filename_stem, li[0][3], bragg_edge)\n",
+    "        if not isinstance(li, list):\n",
+    "            print(\"Input is not a list.\")\n",
+    "            return\n",
+    "        whole_path = os.path.join(output_directory,output_filename)\n",
+    "        if os.path.isfile(whole_path):\n",
+    "            sys.stderr.write(\"ERROR: file {:s} already exists!\\n\".format(output_filename))\n",
+    "            return\n",
+    "\n",
+    "        # TODO replace with np.savetxt\n",
+    "        with open(whole_path, 'w') as output_file:\n",
+    "            for line in li:\n",
+    "                for element in line:\n",
+    "                    if not element == line[-1]:\n",
+    "                        output_file.write(\"{}\\t\".format(element))\n",
+    "                    else:\n",
+    "                        output_file.write(\"{}\\n\".format(element))\n",
+    "        print(\"File {} was created.\".format(output_filename))\n",
+    "        output_filenames.append(output_filename)\n",
+    "    return output_filenames\n",
+    "\n",
+    "tbin_width = 64*2.5\n",
+    "output_filename_table_stem = '{:03d}_table_strain_analysis_{}xy_{}usbin_{}thresh_initintegr'.format(\n",
+    "                        measurement_number,grouping_number,tbin_width,masking_threshold)\n",
+    "\n",
+    "\n",
+    "\n",
+    "write_ws_to_ascii(fit_list, output_filename_table_stem, output_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Generate single tile plot"
```

Comment:
This is running really very slowly. Been waiting for 10 minutes for it to complete. Insufficient memory maybe?